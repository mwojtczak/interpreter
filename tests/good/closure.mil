// closure ma dostęp do zmiennych zadeklarownych w ciele funkcji w której jest zwracany 
// (zmienna k)

function ret_fun(val i, val j) {
	k = 100;
	return function(var a, var b) {
		print(k);
		temp = a + b;
		a = 10;
		b = 20;
		return temp;
	}
}

function main() {
	a = 1;
	b = 2;
	f = ret_fun(a, b);
	print(f(a, b));
	print(a);
	print(b);
}
