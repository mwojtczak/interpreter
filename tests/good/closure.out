100
3
10
20
